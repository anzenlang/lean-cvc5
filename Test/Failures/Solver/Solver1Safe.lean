import Test.Init

namespace cvc5.Safe.Test

def work : IO Unit := Smt.run! .qf_lia do
  Smt.setOption "produce-proofs" "true"

  let n1 ← Smt.declareConst "n1" Int
  let n2 ← Smt.declareConst "n2" Int

  let b ← Smt.declareConst "b" Bool

  let badEq ← Smt.mkEqual n1 b
  let badNot ← Smt.mkNot n1
  let badIte1 ← Smt.mkIte n1 n1 n2
  let badIte2 ← Smt.mkIte b n1 b

  let eq ← Smt.mkEqual n1 n2
  let neq ← Smt.mkNot eq
  let ite ← Smt.mkIte b eq neq

  Smt.assertFormula n1

  Smt.getProof

  Smt.getValue b

  Smt.getValues #[n1, n2]
