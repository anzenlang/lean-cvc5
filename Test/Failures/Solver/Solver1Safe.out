Test/Failures/Solver/Solver1Safe.lean:13:29: error: application type mismatch
  Smt.mkEqual n1 b
argument
  b
has type
  Term Bool : Type
but is expected to have type
  Term Int : Type
Test/Failures/Solver/Solver1Safe.lean:14:25: error: application type mismatch
  Smt.mkNot n1
argument
  n1
has type
  Term Int : Type
but is expected to have type
  Term Bool : Type
Test/Failures/Solver/Solver1Safe.lean:15:26: error: application type mismatch
  Smt.mkIte n1
argument
  n1
has type
  Term Int : Type
but is expected to have type
  Term Bool : Type
Test/Failures/Solver/Solver1Safe.lean:16:31: error: application type mismatch
  Smt.mkIte b n1 b
argument
  b
has type
  Term Bool : Type
but is expected to have type
  Term Int : Type
Test/Failures/Solver/Solver1Safe.lean:22:20: error: application type mismatch
  Smt.assertFormula n1
argument
  n1
has type
  Term Int : Type
but is expected to have type
  Term Bool : Type
Test/Failures/Solver/Solver1Safe.lean:24:2: error: type mismatch
  Smt.getProof
has type
  Smt.UnsatT ?m.1257 (Array Proof) : Type ?u.1256
but is expected to have type
  SmtT IO PUnit : Type
Test/Failures/Solver/Solver1Safe.lean:26:2: error: type mismatch
  Smt.getValue b
has type
  Smt.SatT ?m.1697 (Term Bool) : Type ?u.1696
but is expected to have type
  SmtT IO PUnit : Type
Test/Failures/Solver/Solver1Safe.lean:28:2: error: type mismatch
  Smt.getValues #[n1, n2]
has type
  Smt.SatT ?m.2103 (Array (Term Int)) : Type ?u.2102
but is expected to have type
  SmtT IO Unit : Type